ros/src/computing/perception/semantics/packages/costmap_generator/test/src/test_costmap_generator.cpp

```diff
@@ -0,0 +1,460 @@
+  double expand_polygon_size = 1;
+
+  in_centroid.x = 0;
+  in_centroid.y = 0;
+  in_centroid.z = 0;
+
+  in_corner_point.x = 1;
+  in_corner_point.y = 1;
+  in_corner_point.z = 1;
+
+  geometry_msgs::Point expanded_point = test_obj_.makeExpandedPoint(in_centroid, in_corner_point, expand_polygon_size);
+  double expected_x = 1.707109;
+  double expected_y = 1.707109;
+  const double buffer = 0.01;
+  EXPECT_NEAR(expected_x, expanded_point.x, buffer);
+  EXPECT_NEAR(expected_y, expanded_point.y, buffer);
+}
+
+TEST_F(TestSuite, ChecMakePolygonFromObjectConvexhull)
+{
+  double expand_polygon_size = 0;
+  geometry_msgs::Point32 point;
+  point.x = -1;
+  point.y = -1;
+  point.z = 0;
+  test_obj_.dummy_object_->convex_hull.polygon.points.push_back(point);
+  point.x = 1;
+  point.y = 1;
+  point.z = 0;
+  test_obj_.dummy_object_->convex_hull.polygon.points.push_back(point);
+  point.x = 0;
+  point.y = 2;
+  point.z = 0;
+  test_obj_.dummy_object_->convex_hull.polygon.points.push_back(point);
+
+  grid_map::Polygon polygon =
+  test_obj_.makePolygonFromObjectConvexHull(*test_obj_.dummy_object_,
+                                                expand_polygon_size);
+  double expected_0_x = -1;
+  double expected_1_y =  1;
+  EXPECT_EQ(expected_0_x, polygon[0].x());
+  EXPECT_EQ(expected_1_y, polygon[1].y());
+}
+
+TEST_F(TestSuite, ChecMakePolygonFromObjectConvexhullDofferentHeight)
+{
+  double expand_polygon_size = 0;
+  geometry_msgs::Point32 point;
+  point.x = -1;
+  point.y = -1;
+  point.z = 0;
+  test_obj_.dummy_object_->convex_hull.polygon.points.push_back(point);
+  point.x = -1;
+  point.y = 3;
+  point.z = 0;
+  test_obj_.dummy_object_->convex_hull.polygon.points.push_back(point);
+  point.x = 1;
+  point.y = 1;
+  point.z = 0;
+  test_obj_.dummy_object_->convex_hull.polygon.points.push_back(point);
+  point.x = 0;
+  point.y = 2;
+  point.z = 3;
+  test_obj_.dummy_object_->convex_hull.polygon.points.push_back(point);
+
+  grid_map::Polygon polygon = test_obj_.makePolygonFromObjectConvexHull(
+                                  *test_obj_.dummy_object_,
+                                   expand_polygon_size);
+  int num_vertices = polygon.nVertices();
+  double expected_num_vertices = 3;
+  EXPECT_EQ(expected_num_vertices, num_vertices);
+}
+
+TEST_F(TestSuite, CheckSetCostInPolygon)
+{
+  std::string layer_name = "test";
+  grid_map::Polygon polygon;
+  polygon.setFrameId(layer_name);
+  polygon.addVertex(grid_map::Position(-1, -1));
+  polygon.addVertex(grid_map::Position(1, -1));
+  polygon.addVertex(grid_map::Position(1, 1));
+  polygon.addVertex(grid_map::Position(-1, 1));
+
+
+  float score = 1;
+  test_obj_.setCostInPolygon(polygon, layer_name, score, *test_obj_.dummy_costmap_);
+  float expected_score = 1;
+  EXPECT_EQ(expected_score, test_obj_.dummy_costmap_->atPosition(layer_name, grid_map::Position(0, 0)));
+}
+
+TEST_F(TestSuite, CheckMakeCostmapFromObjects)
+{
+  autoware_msgs::DetectedObject in_object;
```

Comment:
@k0suke-murakami `autoware_msgs::DetectedObject in_object;` are not used any more, can you remove them? I'll approve so you can merge when they are removed

Reply:
Thank you for checking!
I deleted unnecessary `in_object`.